src/huggingface_hub/inference/_common.py

```diff
@@ -161,13 +167,25 @@ def _open_as_binary(
 
 @contextmanager  # type: ignore
 def _open_as_binary(content: Optional[ContentT]) -> Generator[Optional[BinaryT], None, None]:
-    """Open `content` as a binary file, either from a URL, a local path, or raw bytes.
+    """Open `content` as a binary file, either from a URL, a local path, raw bytes, or a PIL Image.
 
     Do nothing if `content` is None,
 
-    TODO: handle a PIL.Image as input
     TODO: handle base64 as input
     """
+    # If content is a PIL Image => convert to bytes
+    if is_pillow_available():
```

Comment:
Can you move this new section down in the method? Something like that: 

```py
    if isinstance(content, str):
        (...)
        yield
        return
    if isinstance(content, Path):
        (...)
        yield
        return  # <- this is new 

    if is_pillow_available():
        (new section)
        yield
        return

    # Otherwise: already a file-like object or None
    yield content
```

Advantage is that if input is a url or path, we won't import PIL at all.